spec/api-menu-spec.js

```diff
@@ -430,3 +430,25 @@ describe('menu module', function () {
     })
   })
 })
+
+describe('MenuItem with custom properties in constructor', function () {
+  it('preserves the custom properties', function () {
+    var template = [{
+      label: 'menu 1',
+      customProp: 'foo',
+      submenu: []
+    }]
+
+    var menu = Menu.buildFromTemplate(template)
+    menu.items[0].submenu.append(new MenuItem({
+      label: 'item 1',
+      customProp: 'bar',
+      overrideProperty: 'oops not allowed'
+    }))
+
+    assert.equal(menu.items[0].customProp, 'foo')
+    assert.equal(menu.items[0].submenu.items[0].label, 'item 1')
+    assert.equal(menu.items[0].submenu.items[0].customProp, 'bar')
+    assert.notEqual(typeof menu.items[0].submenu.items[0].overrideProperty, 'string')
```

Comment:
I wrote it this way to avoid coupling the test to the underlying implementation, but I see your point. 👍